dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -249,63 +282,23 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
+    // 3. From the foreground window.
+    // 4. Use the default assets.
+    Microsoft::Windows::AppNotifications::Helpers::AppNotificationAssets assets{};
 
-    if (hWindow)
+    if (FAILED(RetrieveAssetsFromProcess(assets)) &&
+        FAILED(ToastNotifications_RetrieveAssets_Stub(assets)))
     {
-        // Retrieve DisplayName and IconUri
-        // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
-        // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
-        winrt::com_ptr<IPropertyStore> propertyStore;
-        THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
-
-        wil::unique_prop_variant propVariantDisplayName;
-        // Do not throw in case of failure, default to the filepath approach below as fallback to set a DisplayName.
-        THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
-
-        if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr && wcslen(propVariantDisplayName.pwszVal) > 0)
-        {
-            displayName = propVariantDisplayName.pwszVal;
-        }
-
-        wil::unique_prop_variant propVariantIcon;
-        THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
-
-        THROW_HR_IF_MSG(E_UNEXPECTED, (propVariantIcon.vt == VT_EMPTY || propVariantIcon.pwszVal == nullptr), "Icon is not specified");
-
-        THROW_HR_IF_MSG(E_UNEXPECTED, propVariantIcon.vt != VT_LPWSTR, "Icon should be a valid Unicode string");
-
-        THROW_HR_IF_MSG(E_UNEXPECTED, wcslen(propVariantIcon.pwszVal) == 0, "Icon is an empty string");
-
-        iconFilePath = propVariantIcon.pwszVal;
-
-        // Icon filepaths from Shell APIs have this format: <filepath>,-<index>,
-        // since .ico files can have multiple icons in the same file.
-        // NotificationController doesn't seem to support such format, so let it take the first icon by default.
-        auto iteratorForCommaDelimiter{ iconFilePath.find_first_of(L",") };
-        if (iteratorForCommaDelimiter != std::wstring::npos) // It may or may not have an index, which is fine.
-        {
-            iconFilePath.erase(iteratorForCommaDelimiter);
-        }
-
-        auto iteratorForFileExtension{ iconFilePath.find_first_of(L".") };
-        THROW_HR_IF_MSG(E_UNEXPECTED, iteratorForFileExtension == std::wstring::npos, "You must provide a valid filepath as the app icon.");
-
-        std::wstring iconFileExtension{ iconFilePath.substr(iteratorForFileExtension) };
-        THROW_HR_IF_MSG(E_UNEXPECTED, iconFileExtension != L".ico" && iconFileExtension != L".png",
-            "You must provide a supported file extension as the icon (.ico or .png).");
+        assets.displayName = wil::make_unique_string<wil::unique_cotaskmem_string>(SetDisplayNameBasedOnProcessName().c_str());
```

Comment:
This should be called GetDisplayNameFromProcess.
SetDisplayName* seems incorrect.

Reply:
Or even better. Keep the pattern consistent with above. Call it RetrieveDefaultAssets(assets);

Then do THROW__IF_FAILED(RetrieveDEfaultAssets(assets));